harness/determined/pytorch/deepspeed/dsat/_dsat_search_method.py

```diff
@@ -405,18 +434,51 @@ def __init__(
         )
 
     @abstractmethod
-    def get_new_searcher_ops_list(
+    def get_trials_after_validation_completed(
         self,
         searcher_state: searcher.SearcherState,
-        request_id: uuid.UUID,
-        metric: Optional[Union[float, Dict[str, Any]]] = None,
-    ) -> List[searcher.Operation]:
+        last_trial: DSATTrial,
+        metric: Union[float, Dict[str, Any]],
+    ) -> Iterable[DSATTrial]:
```

Comment:
Thinking about this and other functions in terms of the trials makes a lot more sense. Thanks!